### PR TITLE
POC: Tcp backpressure - DO NOT MERGE!

### PR DESCRIPTION
**Pull request by kgiusti**

### CODE REVIEW

File: src/router_core/delivery.c [hunk 1]
```diff
@@ -712,14 +712,15 @@ static void qdr_delivery_anycast_update_CT(qdr_core_t *core, qdr_delivery_t *dlv
...
+    // If the disposition or extended delivery-state has changed and there is a
+    // peer link, set the disposition/delivery-state of the peer
     // If remote settled, the delivery must be unlinked and freed.
     // If remote settled and there is a peer, the peer shall be settled and
     // unlinked.  It shall not be freed until the connection-side thread
     // settles the PN delivery.
     //
-    if (new_disp != dlv->remote_disposition) {
+    // kag fix me!
+    if (new_disp != dlv->remote_disposition || dlv->remote_state) {
```
Comment (grs) [inline src/router_core/delivery.c:723]: An alternative here would be to test the disposition. E.g. TCP_SEQ_ACK (or PN_RECEIVED) would imply an update required.

File: src/router_core/delivery.c [hunk 2]
```diff
@@ -1246,6 +1247,8 @@ qd_delivery_state_t *qdr_delivery_take_local_delivery_state(qdr_delivery_t *dlv,
 //
 void qdr_delivery_move_delivery_state_CT(qdr_delivery_t *dlv, qdr_delivery_t *peer)
 {
+    /// kag: crap, this may be an issue: possible drop of seq updates!
+
```
Comment (grs) [inline src/router_core/delivery.c:1251]: For TCP_SEQ_ACK (or indeed PN_RECEIVED) it is just a sequence number (or two) to be updated. Perhaps we could update that in place?

File: src/adaptors/tcp_adaptor.c [hunk 3]
```diff
@@ -34,6 +34,11 @@
 #include <inttypes.h>
 #include <stdio.h>
 
+const uint32_t TCP_MAX_CAPACITY = 131072;
+const uint64_t TCP_SEQ_ACK = 999;  // TBD
```
Comment (grs) [inline src/adaptors/tcp_adaptor.c:38]: PN_RECEIVED is very similar to this new type, but has two sequences (we could ignore one or the other in the case of TCP should we wish).

One minor upside of using that is that the protocol trace logging in proton already includes the state.

File: src/router_core/delivery.c (hunk 1 above)
Comment (kgiusti) [reply src/router_core/delivery.c:723]: Good point - I'm going to need to refactor this a bit more.  This POC is exposing a few assumptions we've made regarding disposition propagation that need to be revisited.

**Comment by kgiusti** [comment]
Discarded.  Replaced by fix for DISPATCH-2073

**Pull request closed without merging**